.github/workflows/ci.yml

```diff
@@ -0,0 +1,328 @@
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: ${{ env.key }}-plan-${{ hashFiles('dist-newstyle/cache/plan.json') }}
+        restore-keys: ${{ env.key }}-
+
+    - name: 🛠️ Build Cabal dependencies
+      run: |
+        cabal build all \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}" \
+          --only-dependencies
+
+    - name: 💾 Save Cabal dependencies
+      uses: actions/cache/save@v4
+      if: ${{ !env.ACT && steps.cache-cabal.outputs.cache-hit != 'true' }}
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: ${{ steps.cache-cabal.outputs.cache-primary-key }}
+
+    - name: 🏗️ Build
+      if: ${{ matrix.cabal-tests }}
+      run: |
+        cabal build all \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}"
+
+    - name: 🧪 Test
+      if: ${{ matrix.cabal-tests }}
+      run: |
+        cabal test all \
+          -j1 \
+          --test-show-details=direct \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}"
+      env:
+        TASTY_TIMEOUT: "5m"
+
+    - name: 🛠️ Setup cabal-docspec (Linux)
+      if: ${{ matrix.cabal-tests && runner.os == 'Linux' }}
+      uses: ./.github/actions/setup-cabal-docspec
+
+    - name: 🧪 Test with cabal-docspec (Linux)
+      if: ${{ matrix.cabal-tests && runner.os == 'Linux' }}
+      run: |
+        cabal-docspec \
+          --extra-package lsm-tree:prototypes \
+          -XNumericUnderscores \
+          -XOverloadedStrings \
+          -XTypeApplications
+
+    - name: 🏗️ Build documentation
+      if: ${{ matrix.cabal-documentation }}
+      run: cabal haddock-project
+
+    - name: 📦 Upload documentation
+      if: ${{ matrix.cabal-documentation }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: haddocks-${{ runner.os }}-ghc-${{ matrix.ghc-version || env.DEFAULT_GHC_VERSION }}-cabal-${{ matrix.cabal-version || env.DEFAULT_CABAL_VERSION }}
+        path: haddocks
+        if-no-files-found: error
+        retention-days: 1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macOS-latest", "windows-latest"]
+        ghc-version: ["8.10", "9.2", "9.4", "9.6", "9.8", "9.10", "9.12"]
+        cabal-version: ["3.10"]
+        # The inclusion of these default keys in the matrix ensures that any
+        # entry under include that assigns these keys creates a new matrix entry
+        cabal-flags: [""]
+        cabal-project-file: [""]
+        cabal-documentation: [false]
+      
+        exclude:
+          # GHC 8.10 is not supported on macOS wit ARM CPU
+          - ghc-version: "8.10"
+            os: "macOS-latest"
+
+        include:
+          # Include builds for various special cases
+          - name: "Build on macOS with Intel CPU"
+            os: "macOS-13"
```

Comment:
Testing `ghc-8.10` on MacOS-latest is annoying because it uses an ARM CPU, so that's what the `macOS-13` job was for: testing on Mac with `ghc-8.10`. It's now using the default `ghc-9.6`

Reply:
Oh, I thought it was just general Intel CPU testing. Let me add that back.